mix/ClientModel.cpp

```diff
@@ -168,15 +171,8 @@ void ClientModel::executeSequence(std::vector<TransactionSettings> const& _seque
 					throw std::runtime_error("function " + t.functionId.toStdString() + " not found");
 
 				c.encode(f);
-				for (int p = 0; p < f->parametersList().size(); p++)
-				{
-					QVariableDeclaration* var = (QVariableDeclaration*)f->parametersList().at(p);
-					u256 value = 0;
-					auto v = t.parameterValues.find(var->name());
-					if (v != t.parameterValues.cend())
-						value = v->second;
-					c.encode(var, value);
-				}
+				for (int p = 0; p < t.parameterValues.size(); p++)
+					c.push(t.parameterValues.at(p)->encodeValue());
```

Comment:
There needs to be a check if stored parameter type matches actual parameter type for the contract. The contract could have been edited after the state was configured